scripts/add_electricity.py

```diff
@@ -666,6 +667,42 @@ def attach_hydro(n, costs, ppl):
         )
 
 
+def attach_existing_batteries(n, costs, ppl):
+    """
+    Add existing battery storage units from powerplants.csv to the network.
+    """
+    batteries = ppl.query('carrier == "battery"')
+    if batteries.empty:
+        logger.info("No existing batteries found in powerplants.csv.")
+        return
+
+    _add_missing_carriers_from_costs(n, costs, ["battery"])
+
+    # Remove duplicates and reset index like in attach_hydro
+    batteries = batteries.reset_index(drop=True).rename(index=lambda s: f"{s} battery")
+
+    max_hours = snakemake.params.electricity["max_hours"]["battery"]
```

Comment:
By any chance, aren't `max_hours` readily available in the data extracted by powerplantmatching? Could it be an option to calculate them using some simple empirics by the data we have on hand for the real batteries?